### PR TITLE
WSAS-2030 : persisting module globally engaged status

### PR DESCRIPTION
**Pull request by callkalpa**

### CODE REVIEW

File: components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java [hunk 1]
```diff
@@ -73,15 +79,21 @@
     private static final String RAHAS_MODULE_NAME = "rahas";
     
     private PersistenceFactory pf;
+    
+    private Registry registry;
 
     public ModuleAdminService() throws Exception {
         this.axisConfig = getAxisConfig();
         pf = PersistenceFactory.getInstance(axisConfig);
+        registry = DataHolder.getRegistryService().getConfigSystemRegistry(
```
Comment (Kishanthan) [inline components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java]: We have to do a null check before using a service reference from a dataholder.

File: components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java [hunk 2]
```diff
@@ -73,15 +79,21 @@
...
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId());
     }
 
     public ModuleAdminService(AxisConfiguration ac) throws Exception {
         this.axisConfig = ac;
         pf = PersistenceFactory.getInstance(axisConfig);
+        registry = DataHolder.getRegistryService().getConfigSystemRegistry(
```
Comment (Kishanthan) [inline components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java]: We have to do a null check before using a service reference from dataholder.

Comment (jsdjayanga) [inline components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java]: Assigning a service object to a class variable is not a good practice. Instead get it from the data holder and do a null check before using it.

File: components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java (hunk 1 above)
Comment (callkalpa) [inline components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java]: Thanks Kishanthan,

With https://github.com/wso2/carbon-deployment/pull/135, moved the registry service access code to method level.

File: components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java (hunk 2 above)
Comment (callkalpa) [inline components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java]: Thanks Kishanthan and Jayanga,

With https://github.com/wso2/carbon-deployment/pull/135, moved the registry service access code to method level.

File: components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java [hunk 3]
```diff
@@ -290,6 +303,32 @@ public boolean globallyEngageModule(String moduleId) throws ModuleMgtException {
...
+            if (registry.resourceExists(moduleResourcePath)) {
+                moduleResource = registry.get(moduleResourcePath);
+            } else {
+                moduleResource = registry.newResource();
+            }
+            moduleResource.setProperty(RegistryResources.ModuleProperties.GLOBALLY_ENGAGED,
+                                       Boolean.toString(globallyEngagedStatus));
+            registry.put(moduleResourcePath, moduleResource);
+        } catch (RegistryException e) {
+            log.error("Failed to persist globally engaged status of the module: " + axisModule.getName(), e);
```
Comment (nirmal070125) [inline components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java]: What's the reason for not throwing the exception?

Comment (callkalpa) [inline components/service-mgt/module-mgt/org.wso2.carbon.module.mgt/src/main/java/org/wso2/carbon/module/mgt/service/ModuleAdminService.java]: Thanks Nirmal for pointing the issue. Fixed with https://github.com/wso2/carbon-deployment/pull/146

**Pull request merged**